business/checkers/k8shttproutes/no_host_checker.go

```diff
@@ -0,0 +1,48 @@
+package k8shttproutes
+
+import (
+	"fmt"
+
+	k8s_networking_v1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/models"
+)
+
+type NoHostChecker struct {
+	Namespaces       models.Namespaces
+	K8sHTTPRoute     *k8s_networking_v1alpha2.HTTPRoute
+	RegistryServices []*kubernetes.RegistryService
+}
+
+func (n NoHostChecker) Check() ([]*models.IstioCheck, bool) {
+	validations := make([]*models.IstioCheck, 0)
+	valid := true
+	namespace := n.K8sHTTPRoute.Namespace
+
+	for k, httpRoute := range n.K8sHTTPRoute.Spec.Rules {
+		for i, ref := range httpRoute.BackendRefs {
+			if ref.Kind == nil || string(*ref.Kind) != "Service" {
+				continue
+			}
+			if ref.Namespace != nil && string(*ref.Namespace) != "" {
+				namespace = string(*ref.Namespace)
```

Comment:
I think it would be better to create a local variable inside the for loop for namespace. 
If we set namespace, and for the next iterations the namespace is not compared with the default one. See example (details service do exist in bookinfo namespace): 

![image](https://user-images.githubusercontent.com/49480155/211858483-2a9abc02-79ea-45d1-b2dd-b6f9d1f9de03.png)